File: pinyougou_manager_web/src/main/webapp/js/controller/brandController.js
/*controller的功能,为了继承父控制器,需要$controller域*/
app.controller("brandController", function ($scope, $controller,brandService) {
    //继承父控制器:参数1:父控制器的名称,参数2:固定写法,与父控制器共享$controller
    $controller("baseController",{$scope:$scope});

    //1,是查询数据库,获取所有的brands
    $scope.findAll = function () {
        //没有对象,所以用get就可以,这个是异步请求:直接调用service就可以了
        brandService.findAll().success(function (response) {
            //将请求来的list集合给$scope.list  全局域中的list
            $scope.list = response;
        });
    };




    $scope.findPage = function (pageNum, pageSize) {
        brandService.findPage(pageNum, pageSize).success(function (response) {
            //返回的是当前页面:
            $scope.list = response.rows;
            //总记录数:
            $scope.paginationConf.totalItems = response.total;
        })
    }
    //调用service层的search分页查询**********
    $scope.searchEntity={};

    $scope.search=function (pageNum,pageSize) {
        brandService.search($scope.searchEntity,pageNum,pageSize).success(function (response) {
            $scope.list=response.rows;//当前页结果集
            $scope.paginationConf.totalItems=response.total;//总记录数
        })
    }

    $scope.entity = {};

    /*保存功能:*/
    $scope.save = function () {
        var method={};
        if ($scope.entity.id == null) {
            method = brandService.add($scope.entity);
        } else {
            method = brandService.update($scope.entity);
        }
        //保存而已,没必要写俩啊
        method.success(function (response) {
            if (response.success) {
                $scope.reloadList();  //成功重新加载数据
            } else {
                alert(response.message);
            }
        })


    }
    /*findOneby id*/
    $scope.findOne = function (id) {
        brandService.findOne(id).success(function (response) {
            $scope.entity = response;
        })
    }


    /*删除方法*/
    $scope.del = function () {
        if (confirm("您确定要删除么?")) {
            brandService.del($scope.selectIds).success(function (response) {
                if (response.success) {
                    $scope.reloadList();  //成功重新加载数据
                    $scope.selectIds = [];   //清空数组
                } else {
                    alert(response.message);
                }
            })
        }

    }
})

File: pinyougou_manager_web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.github.pagehelper.PageInfo;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.pojo.TbSpecification;
import com.pinyougou.sellergoods.service.BrandService;

import entity.PageResult;
import entity.Result;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

/*@Controller
@ResponseBody    //保证返回给页面的是json格式:*/
@RestController  //相当于 @Controller + @ResponseBody
@RequestMapping("/brand")
public class BrandController {
    //因为是dubbo去注册中心寻找是否有服务
    @Reference
    private BrandService brandService;

    /**
     * 查询所有brand(品牌)
     * @return
     */
    @RequestMapping("/findAll")
    public List<TbBrand> findAll(){

        return brandService.findAll();
    }
    @RequestMapping("/findPage")
    public PageResult findPage(Integer pageNum, Integer pageSize){
        //因为以后可能扩展其他功能,所以不反回pageInfo
        return brandService.findPage(pageNum,pageSize);

    }
    /*
    分页条件查询:
     */
    @RequestMapping("/search")  //将前端传来的查询条件封装成一个对象,查什么封装成什么
    public PageResult search(@RequestBody TbBrand tbBrand , Integer pageNum, Integer pageSize){

        PageResult result = brandService.search(tbBrand,pageNum,pageSize);
        return result;
    }
    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand tbBrand){
        Result result =null;
       try{
           brandService.save(tbBrand);
           result = new Result(true,"保存成功!");
       }catch(Exception e){
           result = new Result(true,"保存失败!");
        }
       return result;
    }
    @RequestMapping("/findOne")
    public TbBrand findOne(Long id){
        return brandService.findOne(id);
    }
    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand tbBrand){

        try{
            brandService.update(tbBrand);
            return new Result(true,"修改成功!");
        }catch(Exception e){
            return new Result(true,"修改失败!");
        }

    }
    @RequestMapping("/del")
    public Result del(Long[] ids){

        try{
            brandService.delete(ids);
            return new Result(true,"删除成功!");
        }catch(Exception e){
            return new Result(true,"删除失败!");
        }

    }
    @RequestMapping("/selectBrandOptions")
    public List<Map> selectBrandOptions(){
        List<Map> options = brandService.selectBrandOptions();
        return options;
    }

}
